components/automate-cli/cmd/chef-automate/version.go

```diff
@@ -134,6 +164,534 @@ func printUpgradeStatus() error {
+			}
+			v, err := extractVersion(pgPkg.Version, VERSIONREGEX)
+			if err != nil {
+				logrus.Error("ERROR", err)
+				return nil, err
+			}
+			versionMap[ip] = v
+
+		}
+
+		return versionMap, nil
+
+	}
+}
+
+func getPgAuth(sshUtil SSHUtil) (string, string) {
+	output, err := sshUtil.connectAndExecuteCommandOnRemote(CONFIGSHOW, true)
+	if err != nil {
+		logrus.Error("Error in config show", err)
+		return "", ""
+	}
+	config, err := toml.Load(output)
+	if err != nil {
+		logrus.Error("Error in parsing config", err)
+		return "", ""
+	}
+	// retrieve data directly
+	superUser := config.Get("global.v1.external.postgresql.auth.password.superuser").(*toml.Tree)
+	if superUser != nil {
+		userName := superUser.Get("username").(string)
+		password := superUser.Get("password").(string)
+		return userName, password
+	}
+
+	return "", ""
+}
+
+func filterPackage(services *[]models.ServiceDetails, packageName string) *models.ServiceDetails {
+	for _, service := range *services {
+		if strings.HasPrefix(service.ServiceName, packageName) {
+			return &service
+		}
+	}
+	return nil
+}
+
+func getIPAddressesFromFlagOrInfra(infra *AutomateHAInfraDetails) ([]string, []string, []string, []string, *list.List) {
+	errorList := list.New()
+	if VersionCommandFlags.node != "" {
+		automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList := getIpAddressesFromFlag(errorList, infra)
+		return automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList
+	}
+	return getIpAddressesFromInfra(errorList, infra)
+}
+
+func getIpAddressesFromFlag(errorList *list.List, infra *AutomateHAInfraDetails) ([]string, []string, []string, []string, *list.List) {
+	var automateIps, chefServerIps, opensearchIps, postgresqlIps, nodes []string
+	if VersionCommandFlags.isAutomate ||
+		VersionCommandFlags.isChefServer ||
+		VersionCommandFlags.isOpenSearch ||
+		VersionCommandFlags.isPostgresql {
+		nodes = splitIP(
+			VersionCommandFlags.node,
+		)
+		if len(nodes) != 0 {
+			if VersionCommandFlags.isAutomate {
+				automateIps, nodes, errorList = validateIPAddresses(errorList, nodes, automateName, ipAddressError, infra)
+			}
+			if VersionCommandFlags.isChefServer {
+				chefServerIps, nodes, errorList = validateIPAddresses(errorList, nodes, chefServerName, ipAddressError, infra)
+			}
+			if VersionCommandFlags.isOpenSearch {
+				opensearchIps, nodes, errorList = validateIPAddresses(errorList, nodes, opensearchName, ipAddressError, infra)
+			}
+			if VersionCommandFlags.isPostgresql {
+				postgresqlIps, nodes, errorList = validateIPAddresses(errorList, nodes, postgresqlName, ipAddressError, infra)
+			}
+			if len(nodes) != 0 && errorList.Len() == 0 {
+				if VersionCommandFlags.isAutomate ||
+					VersionCommandFlags.isChefServer ||
+					VersionCommandFlags.isOpenSearch ||
+					VersionCommandFlags.isPostgresql {
+					services := []string{}
+					if VersionCommandFlags.isAutomate {
+						services = append(services, "Automate")
```

Comment:
use the constant instead of `Automate`, similarly for Chef Server, OpenSearch, Postgresql as well